### PR TITLE
[fix]Null Pointer reference

### PR DESCRIPTION
**Pull request by Pikachu1412**
- [x] Mark this if you consider it ready to merge
- [ ] I've added tests (optional)
- [ ] I wrote some lines in the [book](https://github.com/radareorg/radare2book) (optional)

**Description**
When as is newed, I add the judgment to check if it is empty, if it is empty, return directly

### CODE REVIEW

File: libr/main/rasm2.c [hunk 1]
```diff
@@ -748,7 +751,9 @@ R_API int r_main_rasm2(int argc, const char *argv[]) {
 	}
 	R_FREE (log_level);
 	RAsmState *as = __as_new ();
-
+	if(!as){
```
Comment (trufae) [inline libr/main/rasm2.c:754]: Add proper spaces
```suggestion
	if (!as) {
```

File: libr/main/rasm2.c [hunk 2]
```diff
@@ -748,7 +751,9 @@ R_API int r_main_rasm2(int argc, const char *argv[]) {
...
+		return 0;
```
Comment (trufae) [inline libr/main/rasm2.c:755]: Return 1 because thats an error
```suggestion
		return 1;
```

File: libr/main/rasm2.c [hunk 3]
```diff
@@ -191,6 +191,9 @@ static void rarch2_list(RAsmState *as, const char *arch) {
 	RListIter *iter, *iter2;
 	const char *feat;
 	PJ *pj = NULL;
+	if(!as){
```
Comment (trufae) [inline libr/main/rasm2.c:194]: Probably better to use r_return_if_fail because that shouldnt happen

**Review by trufae — CHANGES_REQUESTED** [review]

**Comment by Pikachu1412** [comment]
Thank you for your suggestions!




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "radareorg/radare2"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年6月25日(星期天) 晚上6:41
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [radareorg/radare2] [fix]Null Pointer reference (PR #21959)





 
@trufae requested changes on this pull request.
 
 
In libr/main/rasm2.c:
 &gt; @@ -748,7 +751,9 @@ R_API int r_main_rasm2(int argc, const char *argv[]) {  	}  	R_FREE (log_level);  	RAsmState *as = __as_new (); - +	if(!as){  
Add proper spaces
 ⬇️ Suggested change -	if(!as){ +	if (!as) {  
 
In libr/main/rasm2.c:
 &gt; @@ -748,7 +751,9 @@ R_API int r_main_rasm2(int argc, const char *argv[]) {  	}  	R_FREE (log_level);  	RAsmState *as = __as_new (); - +	if(!as){ +		return 0;  
Return 1 because thats an error
 ⬇️ Suggested change -		return 0; +		return 1;  
 
In libr/main/rasm2.c:
 &gt; @@ -191,6 +191,9 @@ static void rarch2_list(RAsmState *as, const char *arch) {  	RListIter *iter, *iter2;  	const char *feat;  	PJ *pj = NULL; +	if(!as){  
Probably better to use r_return_if_fail because that shouldnt happen

File: libr/main/rasm2.c (hunk 3 above)
Comment (Pikachu1412) [reply libr/main/rasm2.c]: There is only one call and it was checked before the call, no need to check again

**Comment by trufae** [comment]
Looks good now! Thanks

**Pull request merged**